common/v2/components/AssetIcon.tsx

```diff
@@ -1,19 +1,22 @@
-import React from 'react';
+import React, { useContext } from 'react';
 import styled from 'styled-components';
-import manifest from 'cryptocurrency-icons/manifest.json';
+import { RatesContext } from 'v2/services';
+import { TUuid } from 'v2/types';
+import genericIcon from 'assets/generic.svg';
 
-import { IAsset, TSymbol } from 'v2/types';
 // Relies on https://github.com/atomiclabs/cryptocurrency-icons using fixed version number through CDN
 // @TODO: We should be using our own sprite served over a trusted CDN
-const baseURL = 'https://cdn.mycryptoapi.com/v1/icons';
+const baseURL = 'https://mycryptoapi.com/api/v1/images';
 
-function buildUrl(symbol: TSymbol) {
-  return `${baseURL}/${symbol.toLowerCase()}.svg`;
+function buildUrl(uuid: string) {
+  return `${baseURL}/${uuid}.png`;
 }
 
-function getIconUrl(symbol: TSymbol) {
-  const curr = manifest.find((c: IAsset) => c.symbol === symbol);
-  return curr ? buildUrl(symbol) : buildUrl('generic' as TSymbol);
+function getIconUrl(uuid: string, assetIconsManifest: any) {
+  const assetIconsManifestEntry =
+    assetIconsManifest && assetIconsManifest[uuid] && assetIconsManifest[uuid].coinGeckoId;
+  const curr = assetIconsManifest ? assetIconsManifestEntry : false;
+  return curr ? buildUrl(uuid) : genericIcon;
```

Comment:
Ramda is perfect for prop drilling such as this.
```
// Since the property is dynamic we create a helper function.
const hasIcon = (uuid) => hasPath([uuid, 'coinGeckoId'])

// It handles the undefined cases.
return hasIcon(uuid)(assetIconsManifest) // Pass the object that we want to examine.
   ? buildUrl(uuid)
   : genericIcon
```